### PR TITLE
Added some cleanup around my expenses component to allow for writing …

### PR DESCRIPTION
**Pull request by Julias0**
…unit tests

### CODE REVIEW

File: src/app/fyle/my-expenses/my-expenses.service.ts [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+
+    const dateFilter = selectedFilters.find((filter) => filter.name === 'Date');
+    if (dateFilter) {
+      generatedFilters.date = dateFilter.value;
+      generatedFilters.customDateStart = dateFilter.associatedData?.startDate;
+      generatedFilters.customDateEnd = dateFilter.associatedData?.endDate;
+    }
+
+    const receiptAttachedFilter = selectedFilters.find((filter) => filter.name === 'Receipts Attached');
+
```
Comment (Dimple16) [inline src/app/fyle/my-expenses/my-expenses.service.ts:215]: we can extract a new method for the below, should avoid a bunch of if condition blocks in the same method

File: src/app/fyle/my-expenses/my-expenses.service.ts [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+        generatedFilters.sortParam = 'tx_org_category';
+        generatedFilters.sortDir = 'asc';
+      } else if (sortBy.value === 'categoryZToA') {
+        generatedFilters.sortParam = 'tx_org_category';
+        generatedFilters.sortDir = 'desc';
+      }
+    }
+  }
+
+  getFilters() {
```
Comment (Dimple16) [inline src/app/fyle/my-expenses/my-expenses.service.ts:276]: returning a list

File: src/app/fyle/my-expenses/my-expenses.service.ts [hunk 3]
```diff
@@ -0,0 +1,541 @@
...
+    }
+
+    if (filter.sortParam && filter.sortDir) {
+      this.addSortToGeneatedFilters(filter, generatedFilters);
+    }
+
+    return generatedFilters;
+  }
+
+  addSortToGeneatedFilters(
```
Comment (Dimple16) [inline src/app/fyle/my-expenses/my-expenses.service.ts:440]: addSortToGeneatedFilters -> typo, `addSortToGeneratedFilters`

**Review by Dimple16 — CHANGES_REQUESTED** [review]

File: src/app/core/services/transaction.service.ts [hunk 4]
```diff
@@ -767,4 +769,172 @@ export class TransactionService {
...
+    const newQueryParamsCopy = cloneDeep(newQueryParams);
+    const stateOrFilter = [];
+
+    if (filters.state) {
+      newQueryParamsCopy.tx_report_id = 'is.null';
+      if (filters.state.includes('READY_TO_REPORT')) {
+        stateOrFilter.push('and(tx_state.in.(COMPLETE),or(tx_policy_amount.is.null,tx_policy_amount.gt.0.0001))');
+      }
+
+      if (filters.state.includes('POLICY_VIOLATED')) {
```
Comment (yksolanki9) [inline src/app/core/services/transaction.service.ts:798]: Create enum for filter state?

**Review by yksolanki9 — CHANGES_REQUESTED** [review]

File: src/app/core/services/transaction.service.ts [hunk 5]
```diff
@@ -767,4 +769,172 @@ export class TransactionService {
...
+      newQueryParamsCopy = this.generateCustomDateParams(newQueryParams, filters);
+    }
+
+    return newQueryParamsCopy;
+  }
+
+  generateCustomDateParams(newQueryParams: any, filters) {
+    const newQueryParamsCopy = cloneDeep(newQueryParams);
+    if (filters.date === DateFilters.custom) {
+      const startDate = filters?.customDateStart?.toISOString();
```
[the remark this answers is not in the document]
Comment (Julias0) [reply src/app/core/services/transaction.service.ts:877]: Since we are passing this as query params, afaik only ISO String is supported

**Review by Dimple16 — APPROVED** [review]

**Review by yksolanki9 — APPROVED** [review]

**Pull request merged**